apps/api/src/app/bridge/e2e/sync.e2e.ts

```diff
@@ -561,93 +558,143 @@ describe('Bridge Sync - /bridge/sync (POST)', async () => {
-
-    const firstWorkflowCountResponse = await workflowsRepository.count({ _environmentId: session.environment._id });
-    expect(firstWorkflowCountResponse).to.equal(1);
-
-    const firstWorkflowResponse = await workflowsRepository.findById(
-      firstSyncResponse.body.data[0]._id,
-      session.environment._id
+    const { workflowId, firstWorkflowResponse, firstSyncResponse } = await storeWorkflow(
+      bridgeServer,
+      session,
+      workflowsRepository
     );
 
-    expect(firstWorkflowResponse).to.be.ok;
-    if (!firstWorkflowResponse) {
-      throw new Error('Workflow not found');
-    }
+    await storeControlValue(session, workflowId, controlValuesRepository, firstWorkflowResponse);
 
-    expect(firstWorkflowResponse.name).to.equal(workflowId);
-    expect(firstWorkflowResponse.type).to.equal(WorkflowTypeEnum.BRIDGE);
-    expect(firstWorkflowResponse.rawData.workflowId).to.equal(workflowId);
-    expect(firstWorkflowResponse.triggers[0].identifier).to.equal(workflowId);
-
-    expect(firstWorkflowResponse.steps.length).to.equal(1);
-    expect(firstWorkflowResponse.steps[0].stepId).to.equal('send-email');
-    expect(firstWorkflowResponse.steps[0]._templateId).to.exist;
-
-    await session.testAgent.put(`/v1/bridge/controls/${workflowId}/send-email`).send({
-      variables: { subject: 'Hello World again' },
-    });
+    const secondWorkflowResponse = await updateWorkflow(
+      session,
+      bridgeServer,
+      workflowsRepository,
+      firstSyncResponse,
+      workflowId
+    );
 
-    const firstControlValueResponse = await controlValuesRepository.find({
-      _environmentId: session.environment._id,
-      _workflowId: firstWorkflowResponse._id,
-    });
+    await validateControlValues(controlValuesRepository, session, secondWorkflowResponse, workflowId);
+  });
+});
 
-    expect(firstControlValueResponse.length).to.equal(1);
-    expect(firstControlValueResponse[0].controls.subject).to.equal('Hello World again');
+async function validateControlValues(
+  controlValuesRepository: ControlValuesRepository,
+  session: UserSession,
+  secondWorkflowResponse: NotificationTemplateEntity,
+  workflowId: string
+) {
+  const secondControlValueResponse = await controlValuesRepository.find({
+    _environmentId: session.environment._id,
+    _workflowId: secondWorkflowResponse._id,
+  });
 
-    const firstStepResponse = await session.testAgent.get(`/v1/bridge/controls/${workflowId}/send-email`);
-    expect(firstStepResponse.body.data.controls.subject).to.equal('Hello World again');
+  expect(secondControlValueResponse.length).to.equal(1);
+  expect(secondControlValueResponse[0].controls.subject).to.equal('Hello World again');
+
+  const secondStepResponse = await session.testAgent.get(`/v1/bridge/controls/${workflowId}/send-email`);
+  expect(secondStepResponse.body.data.controls.subject).to.equal('Hello World again');
+}
+
+async function updateWorkflow(
```

Comment:
Ditto about inlining this function in the tests it belongs. This code contains test steps along with its expectations so the name is misleading.